### DISCUSSION TITLE
[Typing] `Array` type is an alias of `Any`

### DISCUSSION THREAD

**Discussion by JyChang012** [google/flax#2762] [Q&A] [answered]
I see `Array` type in many places is an alias of `Any`. Is there any reason for this? Is it a good idea to use `jax.Array` instead?

**Comment by marcvanzee** [comment, marked as answer]
Up until now, we refrained from aliasing `Array` to anything more specific than `Any` since there are many types of JAX arrays. However, now that `jax.Array` has been introduced we could perhaps following your suggestion.

I notice that JAX themselves are still using `Array = Any` ([example](https://github.com/google/jax/blob/f624b6e7511da4a188dfcf6ac28a9d7c05bbd0f1/jax/_src/lax/windowed_reductions.py#L43)). Perhaps you could file an issue to the JAX repo asking them if they are planning to change this. If so, then we could follow suit.
